server/exllama_kernels/exllama_kernels/cuda_func/q4_matmul.cu

```diff
@@ -85,15 +85,15 @@ __global__ void q4_matmul_kernel
             if constexpr (use_half2)
             {
                 half2 w_scale = w_scales_.item_half2half2(group, w_column);
-                uint32_t w_zero = w_zeros_.item(group, w_column) + 1;
+                uint32_t w_zero = (w_zeros_.item(group, w_column) + 1) & 0x0F;
```

Comment:
the problem is in GPTQ's standard of storing zeros-1 instead of original zeros, ref discussion
https://github.com/AutoGPTQ/AutoGPTQ/pull/484#discussion_r1427449006

Reply:
Oh it's done upstream: https://github.com/AutoGPTQ/AutoGPTQ/blob/main/autogptq_extension/exllama/cuda_func/q4_matmul.cu#L90

Fine by me then.